design-toolkit/components/src/components/pagination/index.tsx

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright 2025 Hypergiant Galactic Systems Inc. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+'use client';
+
+import ChevronLeft from '@accelint/icons/chevron-left';
+import ChevronRight from '@accelint/icons/chevron-right';
+import 'client-only';
+import { range } from 'radashi';
+import { createContext, useContext } from 'react';
+import { Button, ToggleButton } from '../button';
+import { Icon } from '../icon';
+import { PaginationStyles } from './styles';
+import type { ProviderProps } from 'react';
+import type {
+  BasePaginationProps,
+  PaginationNavProps,
+  PageNumberContainerProps as PaginationNumberContainerProps,
+  PaginationPageNumberProps,
+  PaginationRange,
+} from './types';
+
+const { container, button } = PaginationStyles();
+const DEFAULT_MIN_RANGE = 1;
+const DEFAULT_MAX_RANGE = 5;
+
+/**
+ * Return min max range for visible pages. As per our design, we limit
+ * the range of numbers to a spread of 5 maximum, getting the lower and upper bounds.
+ *
+ * @param pageCount - total page count
+ * @param currentPage - current page
+ * @returns - Range of 1 to 5 numbers.
+ */
+export function getPaginationRange(
+  pageCount: number,
+  currentPage: number,
+): PaginationRange {
+  if (
+    !(pageCount && currentPage) ||
+    currentPage > pageCount ||
+    pageCount < 1 ||
+    currentPage < 1
+  ) {
+    return { minRange: 0, maxRange: 0 };
+  }
+
+  // Below max display.
+  if (pageCount < 5) {
```

Comment:
```suggestion
  if (pageCount < DEFAULT_MAX_RANGE) {
```

I see in the ticket requirements we are only doing the 5 page display, so I get why this is here and I don't consider it a blocker -- but every engineering instinct I have tells me not to hard-code these numbers.